### PR TITLE
Docker soft memory limit metric

### PR DESCRIPTION
**Pull request by s-maj**
### What does this PR do?

It brings back https://github.com/DataDog/integrations-core/pull/760
 
### Motivation

Currently, DD supports only hard memory limit metric. I would to know when containers are going above soft limit (wrong limit value, unexpected activity) and incorporate this data into my dashboards / alarms and correlate this with other events.

What is memory reservation:
```
Memory reservation is a kind of memory soft limit that allows for greater sharing of memory. ?
Under normal circumstances, containers can use as much of the memory as needed and are > constrained ?only by the hard limits set with the -m/--memory option. When memory reservation is > set, Docker ?
detects memory contention or low memory and forces containers to restrict their consumption to > a reservation limit.
```
See https://docs.docker.com/engine/reference/run/#specify-custom-cgroups

### Additional Notes

None

### CODE REVIEW

File: pkg/util/docker/docker_util.go [hunk 1]
```diff
@@ -256,7 +256,11 @@ func (d *DockerUtil) Containers(cfg *ContainerListConfig) ([]*Container, error)
 			}
 			container.MemLimit, err = cgroup.MemLimit()
 			if err != nil {
-				log.Debugf("Cgroup cpu limit: %s", err)
+				log.Debugf("Cgroup mem limit: %s", err)
```
Comment (hkaj) [inline pkg/util/docker/docker_util.go]: good catch :)

**Review by hkaj — CHANGES_REQUESTED** [review]
Thanks for the PR @s-maj 🎉 

Could you also add the new metric in this structure please? https://github.com/DataDog/datadog-agent/blob/f54705214be32d2d41a74cde65cbc19acef2eed3/test/integration/corechecks/docker/basemetrics_test.go#L31-L39

Once that's done we'll merge it right away.

**Comment by s-maj** [comment]
@hkaj thanks for the prompt response! 

The only outstanding issue is that docker.mem.limit is presented in GiB where docker.mem.soft_limit is in GB but I can't see how to set unit in the agent. How can I fix this?

**Comment by hkaj** [comment]
@s-maj oh I missed that. This means we forgot to update the docker check's metadata in our backend. Just ran the update job, the units should be fixed now.

**Review by hkaj — DISMISSED** [review]

**Comment by hkaj** [comment]
@s-maj Could you sign the CLA so we can merge this PR? :) Also please let me know if you still see a unit mismatch.

**Comment by s-maj** [comment]
Yeah, something is off with CLA. I think I already signed it (see screenshot below) since I can't see sign button anymore. 
screen shot 2018-05-17 at 15 30 53

**Review by hkaj — APPROVED** [review]

**Pull request merged**